client/signup/steps/user/index.jsx

```diff
@@ -133,6 +135,39 @@ function isOauth2RedirectValid( oauth2Redirect ) {
 		return false;
 	}
 }
+
+/**
+ * Extracts the "No, thanks" redirect URL from query parameters for WooCommerce flows.
+ * This function handles the complexity of parameter nesting that can occur due to
+ * multiple redirects in the Jetpack Connect flow.
+ * @param {Object} query - The query parameters object
+ * @returns {string|null} - The redirect URL if valid, null otherwise
+ */
+function getNoThanksRedirectUrl( query ) {
+	// Check for WooCommerce flows - either explicit parameter or Core Profiler flow
+	const isWooCommerceFlow =
+		query.woo_connect_to_account === '1' || query.from === 'woocommerce-core-profiler';
+
+	if ( ! isWooCommerceFlow ) {
+		return null;
+	}
+
+	// Try to get redirect_after_auth from multiple sources
+	let redirectUrl = query.redirect_after_auth;
+
+	// If not found at top level, try to extract from redirect_to parameter
+	if ( ! redirectUrl && query.redirect_to ) {
+		redirectUrl = getQueryArg( query.redirect_to, 'redirect_after_auth' );
+	}
+
+	// Only return if we have a valid redirect URL
+	if ( typeof redirectUrl === 'string' && isWebUri( redirectUrl ) ) {
+		return redirectUrl;
+	}
+
+	return null;
+}
```

Comment:
Can we extract this out of the User step and move under `wp-login/hooks/`? The reason I'm asking is because we will be moving all the flows to the User step in Stepper soon, so we'll need it there. 

Also, why do we have two `getNoThanksRedirectUrl ` functions - is there a way to abstract this into one?